src/App.js

```diff
@@ -0,0 +1,146 @@
+import React, { useState } from 'react';
+import { Box } from '@mui/material';
+
+import Menu from './components/menu/menu';
+import TaskDialog from './components/dialog/dialog';
+import AddTaskButton from './components/input/addTask';
+import ToDoIcon from './components/icons/todoIcon';
+import SearchTaskFilter from './components/search/searchFilter';
+import TaskRepresentation from './components/tasksManagement/taskRepresentation';
+
+const App = () => {
+  const [tasks, setTasks] = useState([]);
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [taskToEdit, setTaskToEdit] = useState({});
+
+  const handleOpenDialog = () => {
+    setDialogOpen(true);
+  };
+
+  const handleCloseDialog = () => {
+    setDialogOpen(false);
+  };
+
+  const handleSearchChange = (event) => {
+    setSearchQuery(event.target.value);
+  };
+
+  const handleSaveTask = ({ taskName, subject, priority, executionDate, taskIndex }) => {
+    if (JSON.stringify(taskToEdit) === JSON.stringify({})) {
+      setTasks((prev) => 
+      [...prev,
+        { 
+          taskIndex: taskIndex,
+          text: taskName,
+          subject: subject,
+          priority: priority,
+          executionDate: executionDate,
+          completed: false,
+          canShow: true 
```

Comment:
in JavaScript, if you have an object where the keys and values are the same, you can use shorthand to define the object

```suggestion
          taskIndex,
          text: taskName,
          subject,
          priority,
          executionDate,
          completed: false,
          canShow: true 
```

Reply:
Ok